miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -66,6 +66,19 @@ proc.authenticate_account_origin
     # => [storage_offset, ...]
 end
 
+#! Checks that the invocation of a procedure originates from the native account.
+#! This procedure will prevent usage of the account procedures which can mutate the account state 
+#! with the foreign accounts.
+#!
+#! Panics if
+#!   - the invocation of the account procedure does not originate from the native account.
+#!
+#! Stack: []
+#! Output: []
+proc.assert_native_account
+    exec.memory::assert_native_account
+end
```

Comment:
What's the benefit of having this procedure? We can probably just call `exec.memory::assert_native_account` directly, no?

Reply:
Yes, that's my oversight. Initially this procedure was calling `memory::get_native_account_id` and `memory::get_account_id`, but then I moved this entire assert into the `memory.masm`, but forgot to remove the proc from the api. I'll remove this proc.